links/gql_websocket_link/lib/src/link.dart

```diff
@@ -245,6 +251,8 @@ class WebSocketLink extends Link {
             ).catchError(_messagesController.addError);
           });
           _reConnectRequests.clear();
+        } else if (parsedMessage is ConnectionError) {
+          _connectionErrorController.add(parsedMessage);
```

Comment:
@agent3bood Thank you for the review.

As you mentioned, `onError` handler is already forwarding stream exceptions.
But `ConnectionError` seems not considered as an exception. And channel's `listen` handler currently handles `ConnectionAck` but not handle `ConnectionError`. So I added public connection error stream for the use-case handling ConnectionError manually.
https://github.com/gql-dart/gql/blob/master/links/gql_websocket_link/lib/src/link.dart#L237

Additionally, as I described in "First issue", when `_messagesController` got an 'error', all the subscription operations streams are closed now. So they cannot be recovered even after reconnection success.
So I added 277-280 Lines.